src/types/any.rs

```diff
@@ -11,31 +11,30 @@ use std::cell::UnsafeCell;
 use std::cmp::Ordering;
 use std::os::raw::c_int;
 
-/// A Python object with GIL lifetime
+/// Represents any Python object.  
 ///
-/// Represents any Python object.  All Python objects can be cast to `PyAny`.
-/// In addition, if the inner object is an instance of type `T`, we can downcast
-/// `PyAny` into `T`.
+/// It currently only appears as a *reference*, `&PyAny`,
+/// with a lifetime that represents the scope during which the GIL is held.
 ///
-/// `PyAny` is used as a reference with a lifetime that represents that the GIL
-/// is held, therefore its API does not require a `Python<'py>` token.
+/// `PyAny` has some interesting properties, which it shares
+/// with the other [native Python types](crate::types):
 ///
-/// See [the guide](https://pyo3.rs/main/types.html) for an explanation
-/// of the different Python object types.
+/// - It can only be obtained and used while the GIL is held,
+/// therefore its API does not require a [`Python<'py>`](crate::Python) token.
+/// - It can't be used in situations where the GIL is temporarily released,
+/// such as [`Python::allow_threads`](crate::Python::allow_threads)'s closure.
+/// - It can be mutated through any reference.
+/// - It can be converted to the GIL-independent [`Py`]`<`[`PyAny`]`>`,
+/// allowing it to outlive the GIL scope. However, using [`Py`]`<`[`PyAny`]`>`'s API
+/// *does* require a [`Python<'py>`](crate::Python) token.
 ///
-/// # Examples
+/// While basic functionality for interacting with Python objects is available on `PyAny`,
+/// doing useful things with it often requires downcasting `PyAny` to a concrete type.
```

Comment:
That is a bit misleading - you can do a whole lot with the abstract object API.

Reply:
Hmm, that's true, that line isn't really that useful. What about gutting it entirely:

```It can be cast to a concrete type with PyAny::downcast (for native Python types only) and FromPyObject::extract. See their documentation for more information.```

Reply:
:+1: